### PR TITLE
Add deployLib.js and update eTestLib.js

### PR DESCRIPTION
**Pull request by tarieuler**
Completed the following:
-  updated eTestLib.js with smart contract verification logic
-  updated hardhat config to support etherscan api keys for ethereum networks and polygon networks
- added deployLib.js file with smart contract verification logic
- Tested deployment plus verification with Goerli and Mumbai (Polygon testnet) networks.
- function `deployContracts()` has one more optional parameter for verifying the deployed smart contracts. 
- function `deployContracts()` has been left in eTestLib and updated in the eTestLib as it's used within `buildBatch()` function to prevent a cyclic dependency between deployLib.js and eTestLib.js and also to avoid code duplication.
- re-ran all existing unit tests to ensure none using the lib is affecte
- For polygon verification, we might get this error sometimes and need to verify any contracts that come back with this error manually or with the hardhat verification task - https://github.com/NomicFoundation/hardhat/issues/3196

Pending:
- integrate more L2s, e.g., zkSync, Optimisim, Arbitrium

### CODE REVIEW

**Review by kasperpawlowski — CHANGES_REQUESTED** [review]

File: test/lib/eTestLib.js [hunk 1]
```diff
@@ -798,6 +864,9 @@ async function deployContracts(provider, wallets, tokenSetupName) {
     // Euler Contracts
 
     ctx.contracts.eulerGeneralView = await (await ctx.factories.EulerGeneralView.deploy(gitCommit)).deployed();
+    verification.contracts.eulerGeneralView = {
```
[the remark this answers is not in the document]
Comment (tarieuler) [reply test/lib/eTestLib.js:867]: that's a good idea, Thanks. I will resolve it @kasperpawlowski

File: test/lib/deployLib.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        ) {
+            console.log(contract, verification.contracts[contract].address, verification.contracts[contract].args, verification.contracts[contract].contractPath);
+            await verifyContract(verification.contracts[contract].address, verification.contracts[contract].args, verification.contracts[contract].contractPath);
+        }
+    }
+}
+
+async function verifyContract(contractAddress, contractArgs, contractPath = null) {
+    try {
+        if (contractPath === null || contractPath === undefined) {
```
[the remark this answers is not in the document]
Comment (tarieuler) [reply test/lib/deployLib.js:35]: Thanks, I'll resolve @kasperpawlowski

File: test/lib/deployLib.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    console.log("Verifying swap handlers");
+    for (let handler of Object.keys(verification.contracts.swapHandlers)) {
+        console.log(handler, verification.contracts.swapHandlers[handler].address, verification.contracts.swapHandlers[handler].args, verification.contracts.swapHandlers[handler].contractPath);
+        await verifyContract(verification.contracts.swapHandlers[handler].address, verification.contracts.swapHandlers[handler].args, verification.contracts.swapHandlers[handler].contractPath);
+    }
+
+    console.log("Verifying euler contracts");
+    for (let contract of Object.keys(verification.contracts)) {
+        if (
```
[the remark this answers is not in the document]
Comment (tarieuler) [reply test/lib/deployLib.js:22]: Thanks. Will resolve this.

File: test/lib/deployLib.js [hunk 4]
```diff
@@ -1,38 +1,44 @@
...
+            }
         }
     }
+    
 }
 
 async function verifyContract(contractAddress, contractArgs, contractPath = null) {
     try {
-        if (contractPath === null || contractPath === undefined) {
+        if (contractPath) {
```
Comment (kasperpawlowski) [inline test/lib/deployLib.js:41]: actions do not match conditions. either negate the condition or switch actions

**Review by kasperpawlowski — CHANGES_REQUESTED** [review]

File: test/lib/deployLib.js (hunk 4 above)
Comment (tarieuler) [reply test/lib/deployLib.js:41]: thanks for spotting that. All resolved @kasperpawlowski

**Review by kasperpawlowski — APPROVED** [review]

**Pull request merged**